server/api/permission/list/service.js

```diff
@@ -36,7 +36,17 @@ async function listPermissions(db, { q, limit, cursor, scopes }) {
                 $gt: ObjectId(cursor.id),
               },
             }
-          : {}
+          : {},
+        role
+          ? {
+            _id: { $in: role.permissions.map((permission) => ObjectId(permission)) },
```

Comment:
Possible collision with `cursor` prop query. Both `cursor` and `role` add `_id` prop to the match query. I believe my suggestion above fixes this.